pkg/integration/tests/sync/pull_rebase.go

```diff
@@ -22,7 +22,8 @@ var PullRebase = NewIntegrationTest(NewIntegrationTestArgs{
 		shell.SetBranchUpstream("master", "origin/master")
 
 		shell.HardReset("HEAD^^")
-		shell.EmptyCommit("four")
```

Comment:
in older Git, `git pull --rebase` doesn't keep empty commits.

Reply:
This begs the question of what we should do when we have a test that fails only on a single git version, or a subset of our supported versions. We really have three options:
1) update the test to be less picky and pass across all git versions
2) update the code to ensure functionality is the same across all git versions
3) capture the different functionality in the test

When it comes to something like `git pull --rebase`, it doesn't bother me that lazygit behaves differently depending on the git version, because:
* users on an old git version won't be surprised if lazygit does what git already does on the command line
* it would be a pain in the ass to add bespoke code getting the old git versions to behave like new git versions

So for that specific example, I'm fine with option 1 above.

And for the '--no-show-signature' example, option 2 works because it's trivial to do.

But I'd like to have a plan for when we do need to invoke option 3 i.e. we come across functionality that we really do want to test, where older versions behave differently or not at all (e.g. we just show a 'this feature is not supported on your git version, please update' alert panel).

For this, a couple options come to mind:
a) test-level separations: specify in the test which git versions the test is expected to pass on (and create a separate test for the other versions)
b) assertion-level separations: within the test, do something like:
```go
if t.GitVersion("> 2.20") {
  // assert foo
else { 
  // assert bar
}
```

I can imagine both test-level separations and assertion-level separations being needed at some point, but I imagine that assertion-level separations will be more common.

With all that said, what are your thoughts on adding some way to check the git version from within a test, so that we can capture differences across versions?

Reply:
As an experiment, I tried to make the integration test `move_to_earlier_commit.go` pass with any git version (it fails because of `--empty=keep`). It would be possible, but require a bit of work, and I'm not sure it's worth it. On the other hand, not having the test to prove that the feature works at all in earlier git versions would also be a shame. What do you think?

Reply:
A rough version of that experiment is here: https://github.com/stefanhaller/lazygit/commit/2878ef20cbffaa18aac2d757e16e3402453fb73c. Not really pretty...

Reply:
It would be great to make some progress with this. What do you both think about the experiment above?

For an upcoming PR I am in the situation again that I need version-specific integration tests. In this case I need to do a `git rebase -i --update-refs`, which is only available from 2.38 on. I don't see any other way than to disable that test for earlier versions. Not sure how to do that, the `Skip` property seems to accept only literal bools right now. Should we add a new `MinGitVersion` property to `NewIntegrationTestArgs`?

Reply:
@stefanhaller How about changing `Skip` to `func(...) bool`?

Reply:
I reckon we add a new property to `NewIntegrationTestArgs` as @stefanhaller suggests. I'm thinking the field would be `GitVersion` where the value would be a `GitVersionRestriction` struct instance that can be constructed with a method e.g. `GreaterThan("2.1.1")` or `LessThan("2.1.1")`, or Between("2.1.1", "2.1.4") or `Include("2.4.1", "2.4.2", "2.4.3")` or `Exclude(...)`. That should give us the flexibility we want. What do we think about that?

We've got existing code in `pkg/commands/git_commands/version.go` we can make use of for this

Reply:
I added something like this in the PR where I needed it, see https://github.com/jesseduffield/lazygit/pull/2547/commits/01d76d2504364da33003043f8d4352e711a3185d. I only implemented `GreaterThan`, `LessThan`, and `Include` for now; we can add others when we need them, but I kind of doubt we will.

Actually I chose slightly different names, as I think in practice you'll want `GreaterThanOrEqual` in most cases. I found that a bit unwieldy, so I chose `From` and `Before` instead. I'm a bit unhappy about global functions with such generic names, but it does look good at the call site. 😄